examples/counter-next/shared/src/ffi.rs

```diff
@@ -164,3 +164,90 @@ pub mod wasm_ffi {
         }
     }
 }
+
+#[cfg(all(target_os = "wasi", target_env = "p2"))]
+pub mod wasip2 {
+    use crux_core::{Core, bridge::BridgeWithSerializer, type_generation::facet::TypeRegistry};
+    use std::sync::LazyLock;
+
+    use crate::App;
+
+    /// The main interface used by the shell
+    pub struct CoreFFI {
+        core: BridgeWithSerializer<App>,
+    }
+
+    impl CoreFFI {
+        pub fn new() -> Self {
+            let core = BridgeWithSerializer::new(Core::new());
```

Comment:
You _could_ use the new bridge with plug-in FFI format here. 

Not that you'll get much for that, it's just a bit less hassle to deal with the serializers.